### PR TITLE
Update bot label and test logic

### PR DESCRIPTION
**Pull request by Piinks**
This adds:
 - support for more labels to the github bot
 - test exemption for dart fixes
 - test exemption for PRs opened by the flake bot
 - test exemption when only deleting lines of code
 - test exemption for `.md` and `.github` files
 - test requirement for files other than just `.dart`

### CODE REVIEW

**Comment by Piinks** [comment]
CI is blocked on https://github.com/flutter/flutter/issues/89709 for now, but this is not urgent.

File: app_dart/test/request_handlers/github_webhook_test.dart [hunk 1]
```diff
@@ -317,6 +325,74 @@ void main() {
...
+            contains('f: scrolling'));
+      });
+
+      test('integration_test label applied', () {
+        expect(
+            GithubWebhook.getLabelsForFrameworkPath(
+                'dev/integration_tests/web_e2e_tests/test_driver/text_editing_integration.dart'),
+            contains('integration_test'));
+        expect(GithubWebhook.getLabelsForFrameworkPath('packages/integration_test/lib/common.dart'),
+            contains('integration_test'));
```
Comment (Hixie) [inline app_dart/test/request_handlers/github_webhook_test.dart:395]: (this file could be easier to review with some more consistent formatting)

**Review by Hixie — APPROVED** [review]
LGTM

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 2]
```diff
@@ -204,12 +205,19 @@ class GithubWebhook extends RequestHandler<Body> {
           !file.filename.startsWith('dev/bots/')) {
```
Comment (Hixie) [inline app_dart/lib/src/request_handlers/github_webhook.dart:205]: PullRequestFile has a deletionsCount, additionsCount, etc, which would make implementing `A PR that only removes code (no modified or added lines) is exempt from the need to add tests.` pretty easy.

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 3]
```diff
@@ -204,12 +205,19 @@ class GithubWebhook extends RequestHandler<Body> {
...
         needsTests = true;
       }
-      if (file.filename.endsWith('_test.dart')) {
+      if (file.filename.endsWith('_test.dart') ||
+          file.filename.endsWith('.expect') ||
+          file.filename.contains('test_fixes')) {
```
Comment (Hixie) [inline app_dart/lib/src/request_handlers/github_webhook.dart:210]: These two could be easily supported too:
> * A PR that only affect code inside the .github directory is exempt from the need to add tests.
> * A PR that only affects .md files is exempt from the need to add tests.

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 4]
```diff
@@ -204,12 +205,19 @@ class GithubWebhook extends RequestHandler<Body> {
           !file.filename.startsWith('dev/bots/')) {
         needsTests = true;
```
Comment (Hixie) [inline app_dart/lib/src/request_handlers/github_webhook.dart:206]: `A PR that only affects comments (including documentation) is exempt from the need to add tests.` could be supported by looking at the file.patch string, assuming that's the diff for the change.
We'd need to check that every line that starts with a `+` or `-` matches the regexp `^[+-] *//`.

File: app_dart/test/request_handlers/github_webhook_test.dart (hunk 1 above)
Comment (Piinks) [reply app_dart/test/request_handlers/github_webhook_test.dart:395]: I agree, this repo has a formatting script that is enforced though.

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 3 above)
Comment (Piinks) [reply app_dart/lib/src/request_handlers/github_webhook.dart:210]: They already are. Only files ending with `.dart` are marked as `needsTests`. I just think we are all trained now to ask. 😄 
For example: The bot did not flag this `.md`-only change: https://github.com/flutter/flutter/pull/90060

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 2 above)
Comment (Piinks) [reply app_dart/lib/src/request_handlers/github_webhook.dart:205]: Oh nice!

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 4 above)
Comment (Piinks) [reply app_dart/lib/src/request_handlers/github_webhook.dart:206]: This is a bit trickier I think. IIRC the diffs are only available in 100 line chunks, and we already flirt with the github api request limit, but I'll double check this.

Comment (Hixie) [reply app_dart/lib/src/request_handlers/github_webhook.dart:206]: oh i didn't realize we didn't get the whole thing immediately.

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 3 above)
Comment (Hixie) [reply app_dart/lib/src/request_handlers/github_webhook.dart:210]: Hmm, changes to other files should also get tests, not just .dart files...

File: app_dart/test/request_handlers/github_webhook_test.dart (hunk 1 above)
Comment (Hixie) [reply app_dart/test/request_handlers/github_webhook_test.dart:395]: We could... change that? Making the code hard to review is definitely not the purpose of formatting the code...

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 3 above)
Comment (Piinks) [reply app_dart/lib/src/request_handlers/github_webhook.dart:210]: That's what I figures, I'll add this to my list of updates once the cocoon tree is open again.

**Comment by CaseyHillers** [comment]
Cocoon is now accepting pull requests again. To make your PR compatible,

1. Please update the base to point to main (edit in the GitHub UI)
2. Please rebase on tip of tree. There is now null safety support and you'll have to update your PR to be null safe. There may be some rough edges around the initial migration, so feel free to add fixes as you see fit.

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 5]
```diff
@@ -191,25 +191,36 @@ class GithubWebhook extends RequestHandler<Body> {
...
     log!.info('Applying framework repo labels for: owner=${slug.owner} repo=${slug.name} and pr=${pr.number}');
     final Stream<PullRequestFile> files = gitHubClient.pullRequests.listFiles(slug, pr.number!);
+
     final Set<String> labels = <String>{};
     bool hasTests = false;
     bool needsTests = false;
 
     await for (PullRequestFile file in files) {
-      if (file.filename!.endsWith('.dart') &&
+      if (!file.filename!.contains('.gitignore') &&
```
Comment (Piinks) [inline app_dart/lib/src/request_handlers/github_webhook.dart:204]:
```suggestion
      if (!file.filename!.contains('.github') &&
```

**Review by Hixie — APPROVED** [review]
LGTM

**Pull request merged**